cmd/snapctl/main.go

```diff
@@ -57,27 +57,48 @@ func init() {
 	url := flURL.Value
 	ver := flAPIVer.Value
 	secure := false
+	var arg []string
 
 	for idx, a := range os.Args {
 		switch a {
 		case "--url":
-			if err := f1.Parse(os.Args[idx : idx+2]); err == nil {
+			if len(os.Args) >= idx+2 {
+				arg = os.Args[idx : idx+2]
+			} else {
+				break
+			}
+			if err := f1.Parse(arg); err == nil {
```

Comment:
I changed it a little bit to not use an explicit break statement and be shorter/clearer. I don't know that you gain anything by using a function.

The section referenced above is now: 
```go
case "--url":
	if len(os.Args) >= idx+2 {
		if err := f1.Parse(os.Args[idx : idx+2]); err == nil {
			url = *prtURL
		}
	}
```